src/gasman.c

```diff
@@ -2081,54 +2081,75 @@ void CheckMasterPointers( void )
 }
 
 
-/****************************************************************************
-**
-*F  SwapMasterPoint( <bag1>, <bag2> ) . . . swap pointer of <bag1> and <bag2>
-*/
-void SwapMasterPoint (
-    Bag                 bag1,
-    Bag                 bag2 )
+// Swap the master pointers of bag1 and bag2
+//
+// We need to make sure the correct bags are garbage collected, so we always put
+// *both* bags on the ChangedBags linked-list, rather than pick through the
+// exact cases, as it is never incorrect to mark something changed.
```

Comment:
Please don't hit me, but, despite *me* having suggested this wording, I now wonder whether using the term "mark" here might be confusing, because there is also a list of "marked" bugs, and we "mark" bags as part of the mark-and-sweep. So somebody might misinterpret "mark" here? But maybe I am overly nervous here?

An alternative wording here (and later on) might be: "..never incorrect to put a bag into the ChangedBags list" ?

Reply:
I as wondering the same, hence tried to use the "being on the `ChangedBags`" formulation, albeit not consistently.